app/server/lib/MinIOExternalStorage.ts

```diff
@@ -53,8 +53,10 @@ export class MinIOExternalStorage implements ExternalStorage {
       endPoint: string,
       port?: number,
       useSSL?: boolean,
-      accessKey: string,
-      secretKey: string,
+      accessKey?: string,
+      secretKey?: string,
+      credentialsProvider?: any,
```

Comment:
Could you type this property?

Import `IamAwsProvider` like in `app/server/lib/configureMinIOExternalStorage.ts` and then:

```suggestion
      credentialsProvider?: IamAwsProvider,
```